exonum/src/storage/proof_map_index/key.rs

```diff
@@ -462,13 +474,72 @@ impl PartialOrd for ProofPath {
     }
 }
 
+impl ProofPath {
+    /// Creates a compressed binary representation of the given proof path.
+    ///
+    /// # Binary format
+    ///
+    /// - **bits_len** - total length in bits compressed by the leb128 algorithm.
+    /// - **bytes** - non-null bytes of the given ProofPath, i.e. the first `(bits_len + 7) / 8` bytes.
+    pub fn write_compressed(&self, buffer: &mut [u8]) -> usize {
+        let bits_len = u64::from(self.end());
+        let whole_bytes_len = div_ceil!(bits_len, 8) as usize;
+        let key = &self.raw_key()[0..whole_bytes_len];
+
+        let (buffer, bytes_written) = {
+            let mut writer = Cursor::new(buffer);
+            let mut bytes_written = leb128::write::unsigned(&mut writer, bits_len).unwrap();
+            bytes_written += writer.write(&key).unwrap();
+            (writer.into_inner(), bytes_written)
+        };
+        // Cut the bits that lie to the right of the end.
+        let has_bits_in_last_byte = (bits_len % 8) != 0;
```

Comment:
Minor nit: IMO, the following code is slightly more readable:

```suggestion
        let bits_in_last_byte = bits_len % 8;
        if whole_bytes_len > 0 && bits_in_last_byte > 0 {
            let zero_bits_mask = !(255_u8 << bits_in_last_byte);
            buffer[bytes_written - 1] &= zero_bits_mask;
        }
```